heron/common/src/cpp/basics/mempool.cpp

```diff
@@ -21,5 +21,5 @@
 
 // TODO(nlu): get the pool size limit from config
 MemPool<google::protobuf::Message>* __global_protobuf_pool__ =
-                                   new MemPool<google::protobuf::Message>(50 * 1024 * 1024);
+                                   new MemPool<google::protobuf::Message>(512);
```

Comment:
it is config driven. The initial value is 512, and will be overridden here: https://github.com/twitter/heron/pull/2073/files#diff-5ff5ed00c50afd2302f522a9181e50a1R90

As for why initial value cannot be read from config, see the discussion between Sanjeev and I about circular dependency.